From 808335710d20673e5fe78a3980aae5767fe27d0e Mon Sep 17 00:00:00 2001
From: Shreeya Patel <spatel@ciq.com>
Date: Mon, 15 Sep 2025 14:05:15 +0000
Subject: [PATCH] x86/srso: Add SRSO mitigation for Hygon processors

jira VULN-34947
cve CVE-2023-52482
commit-author Pu Wen <puwen@hygon.cn>
commit a5ef7d68cea1344cf524f04981c2b3f80bedbb0d

Add mitigation for the speculative return stack overflow vulnerability
which exists on Hygon processors too.

	Signed-off-by: Pu Wen <puwen@hygon.cn>
	Signed-off-by: Ingo Molnar <mingo@kernel.org>
	Acked-by: Borislav Petkov (AMD) <bp@alien8.de>
	Cc: <stable@vger.kernel.org>
Link: https://lore.kernel.org/r/tencent_4A14812842F104E93AA722EC939483CEFF05@qq.com
(cherry picked from commit a5ef7d68cea1344cf524f04981c2b3f80bedbb0d)
	Signed-off-by: Shreeya Patel <spatel@ciq.com>
---
 arch/x86/kernel/cpu/common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kernel/cpu/common.c b/arch/x86/kernel/cpu/common.c
index ab1b5bef6e81..e6262da39480 100644
--- a/arch/x86/kernel/cpu/common.c
+++ b/arch/x86/kernel/cpu/common.c
@@ -1389,7 +1389,7 @@ static const struct x86_cpu_id cpu_vuln_blacklist[] __initconst = {
 	VULNBL_AMD(0x15, RETBLEED),
 	VULNBL_AMD(0x16, RETBLEED),
 	VULNBL_AMD(0x17, RETBLEED | SMT_RSB | SRSO),
-	VULNBL_HYGON(0x18, RETBLEED | SMT_RSB),
+	VULNBL_HYGON(0x18, RETBLEED | SMT_RSB | SRSO),
 	VULNBL_AMD(0x19, SRSO),
 	{}
 };
-- 
2.47.3

